lib/ansible/modules/notification/matrix.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+# coding: utf-8
+
+# (c) 2018, Jan Christian Grünhage <jan.christian@gruenhage.xyz>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+author: "Jan Christian Grünhage (@jcgruenhage)"
+requirements: [ matrix-client ]
+module: matrix
+short_description: Send notifications to matrix
+description:
+    - This module sends html formatted notifications to matrix rooms.
+version_added: "2.8"
+options:
+    msg_plain:
+        description:
+            - Plain text form of the message to send to matrix, usually markdown
+        required: true
+    msg_html:
+        description:
+            - HTML form of the message to send to matrix
+        required: true
```

Comment:
i would have single 'msg' and then a 'type' option, this follows the pattern of other modules.

Reply:
That would be fit if the message sent contains either a plain text or an html message. Matrix sends both, a plain text variant (usually markdown) and also an HTML variant in the content of a text message event, unless there is no formatting (then the HTML variant is omitted). Because of that, the two different args are required as is.